### ISSUE TITLE
.NET SDK 5.0.200 and 5.0.104 can't load default styles of custom control

### ISSUE THREAD

**Issue by huoyaoyuan** [microsoft/microsoft-ui-xaml#4414]
**Update:**
**Confirmed that there's a problematic CsWinRT version included by SDK 5.0.104/200/201.** Force downgrading it resolves the issue.

**Workarounds**
Option A: in your `global.json`:
```json
{
  "sdk": {
    "version": "5.0.103",
    "rollForward": "disable"
  }
}
```
However, Windows Update can update your installed 5.0.103 SDK to 5.0.104, which is also buggy.

Option B: in your project file, the managed exe project:
```csproj
<ItemsGroup>
  <!-- If you are targeting 18362 -->
  <FrameworkReference Update="Microsoft.Windows.SDK.NET.Ref" RuntimeFrameworkVersion="10.0.18362.13" />

  <!-- Or -->
  <!-- If you are targeting 19041 -->
  <FrameworkReference Update="Microsoft.Windows.SDK.NET.Ref" RuntimeFrameworkVersion="10.0.19041.13" />
</ItemsGroup>
```
.14 are the buggy verions.

-----

**Describe the bug**
I'm surprised that my previously working app showing blank screen. After investigating with Live Visual Tree, the subtree of a templated control is totally empty. I compared with several commits and realized that the only difference should be .NET SDK update.

I'll try to get a repro later. You may also try to reproduce it yourself.
The control in a control library. It's default style is in a separated resource dictionary, merged by Generic.xaml. The control resource dictionary uses x:Bind and has code behind. When I set breakpoint in the code behind constructor, it's not invoked.

**Version Info**

NuGet package version: 

[Microsoft.WinUI 3.0.0-preview3.201113.0]
[Microsoft.WinUI 3.0.0-preview4.210210.4] 

Windows app type:
| UWP              | Win32            |
| :--------------- | :--------------- |
|  | Yes |

| Windows 10 version                  | Saw the problem? |
| :--------------------------------- | :-------------------- |
| Insider Build (xxxxx)              |    |
| October 2020 Update (19042)        | Yes   |
| May 2020 Update (19041)            |    |
| November 2019 Update (18363)       |    |
| May 2019 Update (18362)            |    |
| October 2018 Update (17763)        |    |
| April 2018 Update (17134)          |    |
| Fall Creators Update (16299)       |    |
| Creators Update (15063)            |    |

| Device form factor | Saw the problem? |
| :----------------- | :--------------- |
| Desktop            | Yes |
| Xbox               |  |
| Surface Hub        |  |
| IoT                |  |

**Comment by StephenLPeters** [comment]
@bartekk8 I'm pretty surprised that the .net sdk could affect our resource lookups... do you think that is possible?

@huoyaoyuan is suspect the .net sdk is a red herring.

**Comment by huoyaoyuan** [comment]
Yes, installing back 5.0.103 solves the issue immediately.
However, I want a workaround other than installing old SDK, since there's installer issue in this version.

**Comment by huoyaoyuan** [comment]
Copying the WinRT.Runtime.dll from 5.0.103 output also fixes the issue. There's something broken with CsWinRT.
@manodasanW can you help here?

**Comment by manodasanW** [comment]
@huoyaoyuan Just to confirm I am following, with 5.0.103 your scenario works, but with 5.0.200 it doesn't.  I assume you downloaded 5.0.200 as part of yesterday's release right?  If so, we might be looking at a CsWinRT issue.

Can you provide a repro so I can look into it?  What type of value are you binding to?

**Comment by huoyaoyuan** [comment]
Yes.  
The issue is about templated control. The involved type, if any, could be `System.Type` and `RuntimeType`.

**Comment by huoyaoyuan** [comment]
[SdkTemplateRepro.zip](https://github.com/microsoft/microsoft-ui-xaml/files/6081212/SdkTemplateRepro.zip)
Minimal repro.
It just creating a custom control using IDE template into a black application, without any trick.
Changing the SDK version in global.json to 5.0.200 will cause the custom control not loaded.

**Comment by manodasanW** [comment]
Thanks @huoyaoyuan I was able to get a repro and I see the issue.  I opened a CsWinRT bug for it.

**Comment by huoyaoyuan** [comment]
@anawishnoff Could you highlight this issue and its workaround for a month? It has very wide impact. The fix could only be available after the .NET SDK update at next month.

**Comment by anawishnoff** [comment]
Thanks for filing this and for providing a workaround @huoyaoyuan. Is this an issue specifically with WinUI 3 Preview 4? I can add it to the announcement post.

**Comment by huoyaoyuan** [comment]
@anawishnoff No. It an issue at .NET side. However, it will impact every default setup of VS of version 16.9.0-16.9.1.

**Comment by azchohfi** [comment]
I believe this is also impacting Microsoft.ProjectReunion.

**Comment by angelazhangmsft** [comment]
This issue has been fixed by C#/WinRT 1.1.4 and will be part of an upcoming .NET SDK update - a workaround until the next .NET SDK update for now is a framework reference override with the latest version:

```xml
<ItemGroup>
  <FrameworkReference Update="Microsoft.Windows.SDK.NET.Ref" RuntimeFrameworkVersion="10.0.18362.15" />
  <FrameworkReference Update="Microsoft.Windows.SDK.NET.Ref" TargetingPackVersion="10.0.18362.15" />
</ItemGroup>

```

**Comment by canfucius** [comment]
> 
> 
> This issue has been fixed by C#/WinRT 1.1.4 and will be part of an upcoming .NET SDK update - a workaround until the next .NET SDK update for now is a framework reference override with the latest version:
> 
> ```
> <ItemGroup>
>   <FrameworkReference Update="Microsoft.Windows.SDK.NET.Ref" RuntimeFrameworkVersion="10.0.18362.15" />
>   <FrameworkReference Update="Microsoft.Windows.SDK.NET.Ref" TargetingPackVersion="10.0.18362.15" />
> </ItemGroup>
> ```

Which file exactly do I copy this to?

**Comment by huoyaoyuan** [comment]
Confirmed that fixed by VS 16.9.4 and .NET SDK 5.0.202
